include/configs/ts7100.h

```diff
@@ -0,0 +1,224 @@
+
+/* SPL options in include/configs */
+#include "imx6_spl.h"
+
+/* Env is at the 1MB boundary in emmc boot partition 0 */
+#define CONFIG_SYS_MMC_ENV_DEV  	0 /* mmcblk0 */
+#define CONFIG_SYS_MMC_ENV_PART 	1 /* boot0 */
+
+#define STATUS_LED_RED                  0
+#define STATUS_LED_GREEN                1
+#define STATUS_LED_BLUE                 2
+
+#define STATUS_LED_BIT                  STATUS_LED_RED
+#define STATUS_LED_STATE                STATUS_LED_ON
+#define STATUS_LED_PERIOD               (CONFIG_SYS_HZ / 2)
+
+/* Size of malloc() pool */
+#define CONFIG_SYS_MALLOC_LEN		(16 * SZ_1M)
+
+#define CONFIG_MXC_UART
+#define CONFIG_MXC_UART_BASE		UART1_BASE
+
+#define BOOT_TARGET_DEVICES(func) \
+	func(USB, usb, 0) \
+	func(MMC, mmc, 0) \
+	func(DHCP, dhcp, na)
+#define CONFIG_BOOTCOMMAND \
+	"run distro_bootcmd;"
+
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	"fdt_high=0xffffffff\0" \
+	"initrd_addr=0x83800000\0" \
+	"initrd_high=0xffffffff\0" \
+	"autoload=no\0" \
+	"clearenv=env default -f -a && env save;\0" \
+	"emmc_dev=0\0" \
+	"loadaddr=" __stringify(CONFIG_LOADADDR) "\0" \
+	"fdtaddr=" __stringify(FDT_ADDR_R) "\0" \
+	"kernel_addr_r=" __stringify(CONFIG_LOADADDR) "\0" \
+	"fdt_addr_r=" __stringify(FDT_ADDR_R) "\0" \
+	"ramdisk_addr_r=" __stringify(RAMDISK_ADDR_R) "\0" \
+	"scriptaddr=" __stringify(CONFIG_LOADADDR) "\0" \
+	"pxefile_addr_r=" __stringify(CONFIG_LOADADDR) "\0" \
+	"clearbootcnt=mw.b 50004018 0;\0" \
+	"console=ttymxc0,115200\0" \
+	"altbootcmd=echo taking some recovery action\0" \
+	"chrg_pct=60\0" \
+	"chrg_verb=0\0" \
+	"silochargeon=silabs scaps disable;" \
+		"if test $silopresent = '1';" \
+			"then if test $jpnochrg = 'off';" \
+				"then silab scaps enable;"\
+			"fi;"\
+		"fi;\0" \
+	"silowaitcharge=if test $silopresent = '1';" \
+		"then if test $jpnochrg = 'on';" \
+			"then echo 'NO CHRG jumper is set, not waiting';" \
+			"else scaps wait pct ${chrg_pct};" \
+		"fi;" \
+	"fi;\0" \
+	"usbprod=usb start;" \
+		"if usb storage;" \
+			"then echo Checking USB storage for updates;" \
+			"if load usb 0:1 ${loadaddr} /tsinit.scr;" \
+				"then led green on;" \
+				"source ${loadaddr};" \
+				"led red off;" \
+				"exit;" \
+			"fi;" \
+			"if load usb 0:1 ${loadaddr} /tsinit.scr.uimg;" \
+				"then led green on;" \
+				"source ${loadaddr};" \
+				"led red off;" \
+				"exit;" \
+			"fi;" \
+		"fi;\0" \
+	"emmcboot=echo Booting from the eMMC ...;" \
+		"if load mmc 0:1 ${loadaddr} /boot/boot.ub;" \
+			"then echo Booting from custom /boot/boot.ub;" \
+			"source ${loadaddr};" \
+		"fi;" \
+		"load mmc 0:1 ${fdtaddr} " \
+		  "/boot/imx6ul-ts${model}-${io_model}.dtb;" \
+		"if load mmc 0:1 ${loadaddr} /boot/zImage;" \
+			"then run silowaitcharge;" \
+			"setenv bootargs root=/dev/mmcblk0p1 rootwait rw " \
+			  "cpu_opts=0x${opts} io_opts=0x${io_opts} " \
+			  "io_model=0x${io_model} ${cmdline_append};" \
+			"bootz ${loadaddr} - ${fdtaddr};" \
+		"else echo Failed to load kernel from eMMC;" \
+		"fi;\0" \
```

Comment:
```suggestion
```

These are from the 2016 u-boot an are deprecated with distroboot.

Reply:
The latter has been super-useful, particularly when distroboot breaks. Or a direct boot is wanted. If removed, I think it better to do so under a separate commit that can be put back.

Reply:
Also, there are many other boards still using `emmcboot`, such as the Colibris. One (first in the list below) actually hedges its bets and uses distroboot as a fallback:
```
$ git grep emmcboot  | grep '[^ ]=[^ ]'
configs/apalis-tk1_defconfig:CONFIG_BOOTCOMMAND="run emmcboot; setenv fdtfile ${soc}-${fdt-module}-${fdt_board}.dtb && run distro_bootcmd"
include/configs/apalis-tk1.h:	"emmcboot=run setup; run emmcfinduuid; run set_emmcargs; " \
include/configs/apalis-tk1.h:	"emmcbootpart=1\0" \
include/configs/apalis-tk1.h:	"emmcdtbload=setenv dtbparam; load mmc ${emmcdev}:${emmcbootpart} " \
include/configs/apalis_imx6.h:	"emmcboot=run setup; run emmcfinduuid; run set_emmcargs; " \
include/configs/apalis_imx6.h:	"emmcbootpart=1\0" \
include/configs/apalis_imx6.h:	"emmcdtbload=setenv dtbparam; load mmc ${emmcdev}:${emmcbootpart} " \
include/configs/cl-som-imx7.h:	"emmcbootscript=setenv mmcdev 1; setenv mmcblk 2; run mmcbootscript\0" \
include/configs/cl-som-imx7.h:	"emmcboot=setenv mmcdev 1; setenv mmcblk 2; run mmcboot\0" \
include/configs/cm_t43.h:	"emmcboot=echo Booting from emmc ... && " \
include/configs/colibri_imx6.h:	"emmcboot=run setup; run emmcfinduuid; run set_emmcargs; " \
include/configs/colibri_imx6.h:	"emmcbootpart=1\0" \
include/configs/colibri_imx6.h:	"emmcdtbload=setenv dtbparam; load mmc ${emmcdev}:${emmcbootpart} " \
include/configs/colibri_imx7.h:	"emmcboot=run setup; run emmcfinduuid; run set_emmcargs; " \
include/configs/colibri_imx7.h:	"emmcbootpart=1\0" \
include/configs/nokia_rx51.h:	"emmcboot=setenv mmcnum 1; run trymmcboot\0" \
include/configs/nokia_rx51.h:	"bootmenu_1=Internal eMMC=run emmcboot\0" \
include/configs/opos6uldev.h:	"emmcboot=run initargs; run addmmcargs; "                                               \
include/configs/opos6uldev.h:	"emmcsafeboot=setenv mmcpart 1; setenv mmcroot /dev/mmcblk0p1 ro; run emmcboot;\0"      \
include/configs/uniphier.h:	"emmcboot=mmcsetn && run bootcmd_mmc${mmc_first_dev}\0" \
$ 
```

Reply:
Something else worth noting: Many other platforms (n=332 defconfigs), particularly those omitting `emmcboot`, have `CONFIG_DISTRO_DEFAULTS` in their defconfigs. This config is mentioned in `doc/README.distro`.

Reply:
Removed as suggested.